File: problems/0070_climbing_stairs/0070_climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1] * 2
        for i in range(2, n + 1):
            cur_dp = dp[0] + dp[1]
            dp[1], dp[0] = cur_dp, dp[1]
        return dp[1]

File: problems/0070_climbing_stairs/0070_climbing_stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[2];
        dp[0] = 1;
        dp[1] = 1;
        for (int i = 2; i < n + 1; i++) {
            int cur = dp[0] + dp[1];
            dp[0] = dp[1];
            dp[1] = cur;
        }
        return dp[1];
    }
}